R/deps.R

```diff
@@ -107,6 +108,61 @@ compare_versions <- function(a, b) {
   vapply(seq_along(a), function(i) compare_var(a[[i]], b[[i]]), integer(1))
 }
 
+install_dev_remotes <- function(pkg, ...) {
+  pkg <- as.package(pkg)
+
+  if (is.null(pkg[["remotes"]])) {
+    return()
+  }
+
+  types <- lapply(pkg[["remotes"]], dev_remote_type)
+
+  lapply(types, function(type) type$fun(type$repository, ...))
+}
+
+# Parse the remotes field split into pieces and get install_ functions for each
+# remote type
+dev_remote_type <- function(remotes = "") {
+
+  if (!nchar(remotes)) {
+    return()
+  }
+
+  dev_packages <- trimws(unlist(strsplit(remotes, ",[[:space:]]*")))
+
+  pieces <- strsplit(dev_packages, "|", fixed = TRUE)
+
+  lengths <- vapply(pieces, length, integer(1))
```

Comment:
I feel like this might be simpler if you just iterated once over pieces:

```R
parse_one <- function(parts) {
  if (length(x) == 1) {
    type <- "github"
    repo <- x
   } else if (length(x) == 2) {
      type <- x[1]
      repo <- x[2]
  }  else {
     stop("Malformed...")
  }
  fun <- match.fun(paste0("install_", type))
}
```